zerver/lib/actions.py

```diff
@@ -2883,7 +2883,9 @@ def notify_realm_emoji(realm):
     send_event(event, user_ids)
 
 def do_add_realm_emoji(realm, name, img_url):
-    RealmEmoji(realm=realm, name=name, img_url=img_url).save()
+    emoji = RealmEmoji(realm=realm, name=name, img_url=img_url)
+    emoji.full_clean()
```

Comment:
So, the main problem I have with this pattern is that the `do_foo` functions in this file all have the semantics of executing a particular change under the assumption that all validation has already been done.  Here, we're adding a function using that same naming pattern, but that does the validation itself (rather than expecting the validation to be done in the caller).  So I think we should at least rename this function to `check_add_realm_emoji` for clarity.  

Finally, do you think there should be any validation that the URL actually loads?